src/mlagility/analysis/analysis.py

```diff
@@ -182,6 +182,10 @@ def call_benchit(
                 key=perf.device,
                 value=vars(perf),
             )
+    finally:
```

Comment:
Even better if you replace line 169 with `finally` since we want to save the stats regardless of whether an exception was encountered in groqit.

Reply:
That would give partial progress towards solving #174 and make the code simpler

Reply:
Will be addressed in https://github.com/groq/mlagility/issues/174